system/Helpers/url_helper.php

```diff
@@ -492,23 +492,29 @@ function auto_link($str, $type = 'both', $popup = FALSE): string
 {
 
 	/**
-	 * Prep URL
-	 *
-	 * Simply adds the http:// part if no scheme is included
+	 * Prep URL - Simply adds the http:// part if no scheme is included.
 	 *
+	 * Formerly used URI, but that does not play nicely with URIs missing
+	 * the scheme.
+	 * 
 	 * @param	string	the URL
 	 * @return	string
 	 */
 	function prep_url($str = ''): string
 	{
-		$uri = new \CodeIgniter\HTTP\URI($str);
-
-		if (empty($uri->getScheme()))
-		{
-			$uri->setScheme('http');
-		}
-
-		return (string) $uri;
+		$pos = strpos('//', $str);
+		if ($pos === false)
+			$str = 'http://'.$str;
+
+//		$uri = new \CodeIgniter\HTTP\URI($str);
```

Comment:
if there was no scheme, it treated the authority as a path, injecting a slash. Thus the URI was http:///example.com.
This only showed up in the prep_url testing. It did not appear to be tested for in URI.